src/main/java/org/iac2/service/fixing/plugin/implementaiton/opentoscacontainer/DockerContainerIssueFixingPlugin.java

```diff
@@ -0,0 +1,129 @@
+        }
+
+        // find docker containers and engines
+
+
+        Map<DockerEngine, Collection<DockerContainer>> dockerEngineCollectionMap = Maps.newHashMap();
+
+        instanceModel.getDeploymentModel().getComponents().stream()
+                .filter(c -> c instanceof DockerContainer)
+                .filter(c -> c.getProperties().containsKey("structuralState"))
+                .filter(c -> {
+                    String structuralState = c.getProperties().get("structuralState").getValue();
+                    return structuralState.equals(StructuralState.NOT_EXPECTED.toString()) | structuralState.equals(StructuralState.REMOVED.toString());
+                })
+                .map(c -> (DockerContainer) c)
+                .forEach(c -> {
+                    DockerEngine dockerEngine = this.findDockerEngine(instanceModel.getDeploymentModel().getComponents(), c);
+                    if (dockerEngineCollectionMap.containsKey(dockerEngine)) {
+                        dockerEngineCollectionMap.get(dockerEngine).add(c);
+                    } else {
+                        Collection<DockerContainer> dockerContainerCollection = Lists.newArrayList();
+                        dockerContainerCollection.add(c);
+                        dockerEngineCollectionMap.put(dockerEngine, dockerContainerCollection);
+                    }
+                });
+
+
+        StringBuilder strB = new StringBuilder();
+        strB.append("DockerContainerIssueFixingPlugin Report:").append("\n");
+
+
+        dockerEngineCollectionMap.forEach((dockerEngine, dockerContainers) -> {
+            String dockerEngineUrl = dockerEngine.getProperties().get("DockerEngineURL").getValue();
+
+            if (dockerEngineUrl.contains("host.docker.internal")) {
+                // this is a little dirty, as we use such an URL in the test environment,
+                // we assume this URL is never like this but only a proper URL/IP
+                // => TODO: FIXME
+                dockerEngineUrl = dockerEngineUrl.replace("host.docker.internal", "localhost");
+            }
+            DockerClient dockerClient = Utils.createDockerClient(dockerEngineUrl);
+
+            dockerContainers.forEach(d -> {
+                if (d.getProperties().get("structuralState").getValue().equals(StructuralState.NOT_EXPECTED.toString())) {
+                    // delete the container
+                    String containerId = d.getProperties().get("ContainerID").getValue();
+                    Container container = dockerClient.listContainersCmd().exec().stream().filter(c -> c.getId().equals(containerId)).findFirst().get();
+                    dockerClient.stopContainerCmd(container.getId()).exec();
+                    dockerClient.removeContainerCmd(container.getId()).exec();
+
+                    // remove from model
+                    d.getRelations().forEach(r -> instanceModel.getDeploymentModel().getTopology().removeEdge(r));
```

Comment:
Actually, I don't think this will remove the node from the deployment model. Check the class [here](https://github.com/UST-EDMM/edmm/blob/6de1c497e261eac2400813a7c44124f200b5ff66/edmm-core/src/main/java/io/github/edmm/model/DeploymentModel.java).
The `topology` is a separate object from the `graph` and also from `componentMap`. Removing components and relations from the `topology` will cause inconsistencies within the deployment model object.
As done in other cases, I think removal should be implemented at the `EntityGraph` level (not `DeploymentModel`), then a new `DeploymentModel` instance must be created from the resulting entity graph.
I will play a bit with this since I have got some experience with `EntityGraph`.